### ISSUE TITLE
Position of password instructions

### ISSUE THREAD

**Issue by iadawn** [w3c/w3c-website#250]
The edit password section of the account security page requests exiting password, new password and a confirmation of the new password. The password complexity instructions are included after all these form fields and connected to the new password field using `aria-describedby`.

This may not be the best approach since content in `aria-describedby` cannot be explored using screen readers in the same way as other in-page content. The length of the instructions may also add quite a bit of complexity when using the form. 

<img width="709" alt="image" src="https://user-images.githubusercontent.com/2444840/222106595-680c3515-0455-4e17-a7ca-6259b04e49d9.png">

**Expected behavior**

Move the password instructions above the form and remove `criteria` from the `aria-describedby`.

<img width="709" alt="image" src="https://user-images.githubusercontent.com/2444840/222108515-1109668e-6106-45eb-b784-c78472390b0f.png">

Note there is also a minor text change in the first sentence of the criteria block:

> A password our algorithm deems weak will be rejected

**Comment by iadawn** [comment]
Note that this form also has the problems raised in #226 in it

**Comment by vivienlacourba** [comment]
@deniak could you look at this issue that should exists on both the edit password and password reset pages.

Also @jean-gui noted that the password strength indication (the "`Strong (77%)`" text in the above screenshot) should probably be moved right under the first "`New Password`" field (which might require some change to that Symfony widget).

**Comment by iadawn** [comment]
> Also @jean-gui noted that the password strength indication (the "`Strong (77%)`" text in the above screenshot) should probably be moved right under the first "`New Password`" field (which might require some change to that Symfony widget).

Moving this would help screen magnifier users as it would more closely associate the content. This isn't 100% but it would help. For screen reader type software the strength indication is included in a live region so all is good for that.

**Comment by deniak** [comment]
@iadawn, on the [account request page](https://beta.w3.org/account/request/), we use a `span.field-hint` to display the password requirements:
```
<span id="hint-w3c_accountsbundle_accountrequeststep1_plainPassword_first" class="field-hint">Passwords <strong>must</strong> be at least 8 characters long and <strong>should</strong> mix uppercase, lowercase, numbers and special characters.</span>
```



I can do the same on the edit password form if you confirm it will work for screen readers. This will also reduce the length of the instructions.
<img width="746" alt="Screenshot 2023-03-03 at 08 57 41" src="https://user-images.githubusercontent.com/1696128/222635169-7d5a43b9-967e-43b5-af8b-19a6da9caa96.png">

**Comment by iadawn** [comment]
Looks much better. I note from the account request page that the `<span>` is connected to the `<label>` using `aria-describedby`. All good :)

A couple of minor quibbles with the text, partly to reduce the length and partly to make it clearer.

I might suggest:

> *Must* be at least 8 characters long and include uppercase, lowercase, numbers and special characters.

Not everyone will distinguish between 'must' and 'should' so may as well remove it for clarity. Also, this is associated with the 'Password' field so probably doesn't need to start with 'Passwords'.

**Comment by deniak** [comment]
@iadawn thanks for the suggestion. We'll implement it.
One thing we aren't sure is how to position the password strength indication. As discussed above, I'll move it after the first password field. However, this is added by javascript and the second input will move because of the addition of that new element. Is it acceptable or do we need to find another position for the strength notice?

**Comment by iadawn** [comment]
Do you mean in the same place as for the account request page?

<img width="455" alt="image" src="https://user-images.githubusercontent.com/2444840/223423814-feeab1ef-0a92-483a-851f-165a6ca08a80.png">

If so then that is fine. The proximity will help screen magnifiers and the live region already supports the likes of screen readers

**Comment by deniak** [comment]
What I mean is that the strength notice creates a small gap between the 2 fields while the user is typing (cf screenshots)

<img width="525" alt="Screenshot 2023-03-07 at 17 07 43" src="https://user-images.githubusercontent.com/1696128/223431656-bca0aa96-e04e-4ae8-a9db-9bafd9909008.png">
<img width="502" alt="Screenshot 2023-03-07 at 17 07 57" src="https://user-images.githubusercontent.com/1696128/223431677-43aa36e8-8f26-4215-8e2c-96abeb09acdf.png">

If that's acceptable, then we'll leave it as is

**Comment by deniak** [comment]
Thanks for the inputs @iadawn, the fix has now been deployed
